packages/atomic-react/src/components/SearchInterfaceWrapper.tsx

```diff
@@ -0,0 +1,51 @@
+import React, {useEffect, useRef} from 'react';
+import type {JSX} from '@coveo/atomic';
+import {
+  buildSearchEngine,
+  getSampleSearchEngineConfiguration,
+} from '@coveo/atomic/headless';
+import {AtomicSearchInterface} from './stencil-generated/index';
+
+type ExecuteSearch = HTMLAtomicSearchInterfaceElement['executeFirstSearch'];
+interface WrapperProps extends JSX.AtomicSearchInterface {
+  onReady?: (executeFirstSearch: ExecuteSearch) => Promise<void>;
+  theme?: string | 'none';
+}
+
+const DefaultProps: Required<
+  Pick<WrapperProps, 'onReady' | 'theme' | 'engine'>
+> = {
+  onReady: (executeFirstSearch) => {
+    return executeFirstSearch();
+  },
+  theme: 'coveo',
+  engine: buildSearchEngine({
```

Comment:
If somebody decides to initialize their own engine by passing it as a prop, would this display a second store in the redux dev tools?

Reply:
Hum probably... Should only build the engine when it's not passed in as a prop then.

Good catch.